### PR TITLE
Cherry-pick fix indefinite stuck Pending pod on a deleted node

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
Cherry-pick https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/139

### CODE REVIEW

**Comment by jsafrane** [comment]
/retest

`git fetch https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner.git v8.0.1` - it should fetch from https://github.com/sunnylovestiramisu/sig-storage-lib-external-provisioner/tree/v8.0.1, not kubernetes-sigs

**Comment by jsafrane** [comment]
This is the failure that matters:

> --- FAIL: TestTopologyParams (1.01s)
    --- FAIL: TestTopologyParams/provision_with_selected_node,_but_node_does_not_exist (0.20s)
        controller_test.go:935: did not expect a Provision() call but got at least 1

**Comment by jsafrane** [comment]
/retest

I've no clue what's wrong with gofmt. It runs [verify-gofmt.sh](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/repo-infra/verify/go-tools/verify-gofmt.sh) in container `golang:1.19`. And it complains about files that are not part of this PR.

**Comment by sunnylovestiramisu** [comment]
Fixed the unit test but the [verify-gofmt.sh](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/repo-infra/verify/go-tools/verify-gofmt.sh) is still failing

**Comment by sunnylovestiramisu** [comment]
Now it is complaining about:

```
PATH=$(go env GOPATH)/bin:$PATH repo-infra/verify/verify-boilerplate.sh
/usr/bin/env: 'python': No such file or directory
```

**Comment by sunnylovestiramisu** [comment]
There is no change in k/k's [verify-boilerplate.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-boilerplate.sh)

File: repo-infra/verify/verify-boilerplate.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ cleanup() {
 }
 
 pushd "${boilerDir}" >/dev/null
-if ! python -m unittest boilerplate_test 2>"${unitTestOut}"; then
+if ! python3 -m unittest boilerplate_test 2>"${unitTestOut}"; then
```
Comment (msau42) [inline repo-infra/verify/verify-boilerplate.sh:41]: You may need to change this to directly use python 3: https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/017c935a0c11cf9e572345600a80733dbf851081/repo-infra/verify/boilerplate/boilerplate.py#L1

Or we need to update the container image used for the job to install "python-is-python3": https://askubuntu.com/questions/942930/usr-bin-env-python-no-such-file-or-directory

Comment (sunnylovestiramisu) [reply repo-infra/verify/verify-boilerplate.sh:41]: I tried change this to python3, and did ``make verify`` locally, it still failed with:

```
/usr/bin/env: ‘python’: No such file or directory
make: *** [Makefile:25: verify] Error 127
```

Comment (sunnylovestiramisu) [reply repo-infra/verify/verify-boilerplate.sh:41]: The job definition is:

```
  - name: pull-sig-storage-lib-external-provisioner-unit
    always_run: true
    decorate: true
    path_alias: sigs.k8s.io/sig-storage-lib-external-provisioner
    annotations:
      testgrid-dashboards: sig-storage-lib-external-provisioner
      testgrid-tab-name: unit
      description: Unit tests in sig-storage-lib-external-provisioner repo.
    spec:
      containers:
      - image: golang:1.19
        command:
        - make
        args:
        - test
```

**Pull request merged**